### DISCUSSION TITLE
`set_message_received_handler` to capture receipt address

### DISCUSSION THREAD

**Discussion by Roban10** [python-smpplib/python-smpplib#181] [Q&A]
How can we get destination address in `deliver_sm`? I am able to capture sms data ans source address only.

**Comment by eigenein** [comment]
Hi @Roban10,

Have you checked [`destination_addr`](https://github.com/python-smpplib/python-smpplib/blob/355713ad394148e66cde4ce5a37b45ab6d60ac41/smpplib/command.py#L761-L763)?

Please, don't create duplicate discussions.

**Reply by Roban10** [reply]
Thank you. I am able to capture the destination_addr now. But there was an error 'broken pdu' after running for sometime. 

```
 client.listen( auto_send_enquire_link= False)
  File "E:\Users\Test\PycharmProjects\untitled\venv\lib\site-packages\smpplib\client.py", line 403, in listen
    self.read_once(ignore_error_codes, auto_send_enquire_link)
  File "E:\Users\Test\PycharmProjects\untitled\venv\lib\site-packages\smpplib\client.py", line 357, in read_once
    pdu = self.read_pdu()
  File "E:\Users\Test\PycharmProjects\untitled\venv\lib\site-packages\smpplib\client.py", line 235, in read_pdu
    raw_len = self._socket.recv(4)
socket.timeout: timed out
<smpplib.client.Client object at 0x000000000231CBA8> was not closed
```

**Comment by Roban10** [comment]
How can we ignore the broken pdu. Due to broken PDU error, the connection with SMSC is automatically disconnect,  and cause loss of incoming sms.
